narwhals/_arrow/series.py

```diff
@@ -584,6 +584,162 @@ def to_string(self, format: str) -> ArrowSeries:  # noqa: A002
+        )
+
+    def microsecond(self: Self) -> ArrowSeries:
+        pc = get_pyarrow_compute()
+        arr = self._arrow_series._native_series
+        result = pc.add(pc.multiply(pc.millisecond(arr), 1000), pc.microsecond(arr))
+
+        return self._arrow_series._from_native_series(result)
+
+    def nanosecond(self: Self) -> ArrowSeries:
+        pc = get_pyarrow_compute()
+        pc = get_pyarrow_compute()
+        arr = self._arrow_series._native_series
+        result = pc.add(
+            pc.multiply(self.microsecond()._native_series, 1000), pc.nanosecond(arr)
+        )
+        return self._arrow_series._from_native_series(result)
+
+    def ordinal_day(self: Self) -> ArrowSeries:
+        pc = get_pyarrow_compute()
+        return self._arrow_series._from_native_series(
+            pc.day_of_year(self._arrow_series._native_series)
+        )
+
+    def total_minutes(self: Self) -> ArrowSeries:
+        pa = get_pyarrow()
+        pc = get_pyarrow_compute()
+        arr = self._arrow_series._native_series
+        unit = arr.type.unit
+
+        unit_to_minutes_factor = {
+            "ms": 60 * 1e3,  # milli
+            "us": 60 * 1e6,  # micro
+            "ns": 60 * 1e9,  # nano
+        }
+
+        factor = pa.scalar(unit_to_minutes_factor[unit], type=pa.int64())
+        return self._arrow_series._from_native_series(
+            pc.cast(pc.divide(arr, factor), pa.int64())
+        )
+
+    def total_seconds(self: Self) -> ArrowSeries:
+        pa = get_pyarrow()
+        pc = get_pyarrow_compute()
+        arr = self._arrow_series._native_series
+        unit = arr.type.unit
+
+        unit_to_seconds_factor = {
+            "ms": 1e3,  # milli
+            "us": 1e6,  # micro
+            "ns": 1e9,  # nano
+        }
+        factor = pa.scalar(unit_to_seconds_factor[unit], type=pa.int64())
+
+        return self._arrow_series._from_native_series(
+            pc.cast(pc.divide(arr, factor), pa.int64())
+        )
+
+    def total_milliseconds(self: Self) -> ArrowSeries:
+        pa = get_pyarrow()
+        pc = get_pyarrow_compute()
+        arr = self._arrow_series._native_series
+        unit = arr.type.unit
+
+        unit_to_milli_factor = {
+            "ms": 1,  # milli
+            "us": 1e3,  # micro
+            "ns": 1e6,  # nano
+        }
+
+        factor = pa.scalar(unit_to_milli_factor[unit], type=pa.int64())
+
+        return self._arrow_series._from_native_series(
+            pc.cast(pc.divide(arr, factor), pa.int64())
+        )
+
+    def total_microseconds(self: Self) -> ArrowSeries:
+        pa = get_pyarrow()
+        pc = get_pyarrow_compute()
+        arr = self._arrow_series._native_series
+        unit = arr.type.unit
+
+        unit_to_micro_factor = {
+            "ms": 1e3,  # milli
+            "us": 1,  # micro
+            "ns": 1e3,  # nano
+        }
+
+        factor = pa.scalar(unit_to_micro_factor[unit], type=pa.int64())
+
+        if unit == "ms":  # pragma: no cover
```

Comment:
is it possible to make a test which hits this? maybe it could be a separate test, in which you specifically instantiate a pyarrow array with 'ms' unit, rather than the parametrised one